go/chat/crypto_doc.md

```diff
@@ -0,0 +1,180 @@
+that the signature *itself* is private. Even though the server knows who's
+talking to whom, because it's delivering all the messages, it's better that it
+can't *prove* what it knows.
+
+### Key Handling
+
+The details in this section are shared with KBFS and the rest of the Keybase
+app.
+
+The encryption key is 32-random bytes. It's shared by everyone in the chat, and
+it's the same key that they use to encrypt files in the private KBFS folder
+that those people share.
+
+The signing key is a 32/64-byte public/private keypair, which is unique to each
+device. This is the same keypair that your device uses to add new signatures to
+your account's signature chain, like when you prove new identities or add other
+devices.
+
+Secret keys are stored on disk, in a file encrypted with
+[TripleSec](https://keybase.io/triplesec/), with a key derived from the user's
+Keybase password. While the user is logged in to Keybase, that key stored on
+disk in plain text (see tradeoffs below), or in the system keyring on OSX. When
+the user logs out of Keybase, that key is deleted, and secret keys are
+unreadable until the next login.
+
+## Limitations and Tradeoffs
+
+### Forward Secrecy
+
+**Tradeoff:** Messages on the server don't have forward secrecy. That is, the
+keys to read them still exist on your devices, and they never get deleted. That
+means that someone who steals your device might be able to read your old
+messages.
+
+**Reason:** This is necessary for users to read their message history on new
+devices. Forward secrecy also gets into a gray area when you allow multiple
+devices on an account. (If you haven't turned on your other phone in 6 months,
+and you keep sending messages to it, the ephemeral keys on that phone aren't
+very ephemeral anymore.) Finally, deleting keys doesn't help you much if you
+keep decrypted message history on the same device. Note that we do generate new
+encryption keys when you remove a device, so a removed device can't decrypt
+messages that were sent after it got removed.
+
+### Repudiability
+
+**Tradeoff:** Chat messages don't have repudiability. That is, if Alice sends a
+message to Bob, it's possible for Bob to prove to other people that Alice sent
+it.
+
+**Reason:** This is a side effect of using signing keys to authenticate
+messages. Repudiable authentication works well in a chat between two people,
+and it's even cheaper than signing. But it doesn't scale well to group chats
+with tons of people, because the sender needs to authenticate each message
+separately for every other person in the group. It also makes it difficult to
+add new people to an existing group (which we might support in the future),
+because old messages can't be reauthenticated until all the senders come
+online.
+
+### History Integrity
+
+**Tradeoff:** It's possible for the server to change history in specific ways,
+like delaying messages or reordering senders who don't know about each other
+yet.
```

Comment:
It wasn't obvious to me what "reordering senders who don't know about each other yet" means.